src/governance-validators.cpp

```diff
@@ -0,0 +1,349 @@
+        fJSONValid = true;
+    }
+    catch(std::exception& e) {
+        strErrorMessages += std::string(e.what()) + std::string(";");
+    }
+    catch(...) {
+        strErrorMessages += "Unknown exception;";
+    }
+}
+
+bool CProposalValidator::GetDataValue(const std::string& strKey, std::string& strValue)
+{
+    bool fOK = false;
+    try  {
+        strValue = objJSON[strKey].get_str();
+        fOK = true;
+    }
+    catch(std::exception& e) {
+        strErrorMessages += std::string(e.what()) + std::string(";");
+    }
+    catch(...) {
+        strErrorMessages += "Unknown exception;";
+    }
+    return fOK;
+}
+
+bool CProposalValidator::GetDataValue(const std::string& strKey, int64_t& nValue)
+{
+    bool fOK = false;
+    try  {
+        const UniValue uValue = objJSON[strKey];
+        switch(uValue.getType()) {
+        case UniValue::VNUM:
+            nValue = uValue.get_int64();
+            fOK = true;
+            break;
+        case UniValue::VSTR:
+        {
+            std::istringstream istr(uValue.get_str());
+            istr >> nValue;
+            fOK = ! istr.fail();
+        }
+        break;
+        default:
+            break;
+        }
+    }
+    catch(std::exception& e) {
+        strErrorMessages += std::string(e.what()) + std::string(";");
+    }
+    catch(...) {
+        strErrorMessages += "Unknown exception;";
+    }
+    return fOK;
+}
+
+bool CProposalValidator::GetDataValue(const std::string& strKey, double& dValue)
+{
+    bool fOK = false;
+    try  {
+        const UniValue uValue = objJSON[strKey];
+        switch(uValue.getType()) {
+        case UniValue::VNUM:
+            dValue = uValue.get_real();
+            fOK = true;
+            break;
+        case UniValue::VSTR:
+        {
+            std::istringstream istr(uValue.get_str());
+            istr >> dValue;
+            fOK = ! istr.fail();
+        }
+        break;
+        default:
+            break;
+        }
+    }
+    catch(std::exception& e) {
+        strErrorMessages += std::string(e.what()) + std::string(";");
+    }
+    catch(...) {
+        strErrorMessages += "Unknown exception;";
+    }
+    return fOK;
+}
+
+std::string CProposalValidator::StripWhitespace(const std::string& strIn)
+{
+    static const std::string strWhitespace = " \f\n\r\t\v";
+
+    std::string::size_type nStart = strIn.find_first_not_of(strWhitespace);
+    std::string::size_type nEnd = strIn.find_last_not_of(strWhitespace);
+
+    if((nStart == std::string::npos) || (nEnd == std::string::npos)) {
+        return std::string();
+    }
+
+    return strIn.substr(nStart, nEnd - nStart + 1);
+}
+
+/*
+  The purpose of this function is to replicate the behavior of the
+  Python urlparse function used by sentinel (urlparse.py).  This function
+  should return false whenever urlparse raises an exception and true
+  otherwise.
+ */
+bool CProposalValidator::CheckURL(const std::string& strURLIn)
+{
+    std::string strRest(strURLIn);
```

Comment:
As this is copying the function argument and is its only use, this std::string could be made the function argument.
'value semantics' also allows an r-value string argument to be moved in, avoiding the copy allocation.